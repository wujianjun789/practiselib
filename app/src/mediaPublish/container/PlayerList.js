/**
 * Created by a on 2017/10/17.
 */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Content from '../../components/Content';
import Select from '../../components/Select';
import SearchText from '../../components/SearchText';
import Table from '../../components/Table';
import Page from '../../components/Page'

import PlayerListItem from '../component/PlayerListItem';
import PlayerListPopup from '../component/PlayerListPopup';
import ConfirmPopup from '../../components/ConfirmPopup';

import { overlayerShow, overlayerHide } from '../../common/actions/overlayer'
import Immutable from 'immutable';
import { getObjectByKey } from '../../util/algorithm'

import { FormattedMessage, injectIntl } from 'react-intl';

export class PlayerList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            type: Immutable.fromJS({
                list: [{ id: 1, value: `${this.props.intl.formatMessage({ id: 'mediaPublish.playType' })} 1` },
                { id: 2, value: `${this.props.intl.formatMessage({ id: 'mediaPublish.playType' })} 2` }],
                index: 0, value: `${this.props.intl.formatMessage({ id: 'mediaPublish.playType' })} 1`
            }),
            search: Immutable.fromJS({ placeholder: this.props.intl.formatMessage({id:'mediaPublish.inputSchemeName'}), value: '' }),
            page: Immutable.fromJS({
                pageSize: 10,
                current: 1,
                total: 2
            }),
            data: Immutable.fromJS([
                { id: 1, icon: "", name: `${this.props.intl.formatMessage({id:'mediaPublish.playPlan'})} 1`, resolution: "1920X1080", width: 1920, height: 1080 },
                { id: 2, icon: "", name: `${this.props.intl.formatMessage({id:'mediaPublish.playPlan'})} 2`, resolution: "1920X1080", width: 1920, height: 1080 }
            ])
        }

        this.columns = [{ id: 1, field: "name", title:this.props.intl.formatMessage({id:'mediaPublish.schemeName'})},
             { id: 2, field: "resolution", title: this.props.intl.formatMessage({id:'mediaPublish.resolution'})}];

        this.typeChange = this.typeChange.bind(this);
        this.searchChange = this.searchChange.bind(this);
        this.searchSubmit = this.searchSubmit.bind(this);
        this.pageChange = this.pageChange.bind(this);
        this.updatePage = this.updatePage.bind(this);

        this.addHandler = this.addHandler.bind(this);
        this.publishHandler = this.publishHandler.bind(this);
        this.funHandler = this.funHandler.bind(this);
        this.editHandler = this.editHandler.bind(this);
        this.removeHandler = this.removeHandler.bind(this);

        this.requestSearch = this.requestSearch.bind(this);
    }

    requestSearch() {

    }

    typeChange(selectIndex) {
        this.state.type = this.state.type.update("index", v => selectIndex);
        this.setState({ type: this.state.type.update("value", v => this.state.type.getIn(["list", selectIndex, "value"])) }, () => {
            this.requestSearch();
        });
    }

    searchChange(value) {
        this.setState({ search: this.state.search.update("value", v => value) });
    }

    searchSubmit() {
        this.updatePage(1);
    }

    pageChange(current, pageSize) {
        this.updatePage(current);
    }

    updatePage(current) {
        this.setState({ page: this.state.page.update("current", v => current) }, () => {
            this.requestSearch();
        });
    }

    addHandler() {
        const { actions } = this.props;
        const data = {
            playerId: '',
            playerName: '',
            width: 1920,
            height: 1080
        }

        actions.overlayerShow(<PlayerListPopup intl={this.props.intl} title={this.props.intl.formatMessage({id:'mediaPublish.addPlan'})} data={data}

            onCancel={() => { actions.overlayerHide() }} onConfirm={(state) => {
                actions.overlayerHide();
                const id = parseInt(Math.random() * 30);
                let item = { id: id, icon: "", name: state.playerName, resolution: state.width + "X" + state.height, width: state.width, height: state.height }
                this.setState({ data: this.state.data.push(Immutable.fromJS(item)) }, () => { console.log('size:', this.state.data.size) })
                this.props.router.push({
                    pathname: "/mediaPublish/playerArea",
                    state: {
                        item: item
                    }
                });
                // this.props.router.push('/playerArea');
            }} />)

    }

    publishHandler(id) {

    }

    funHandler(id) {

    }

    editHandler(id) {
        console.log("edit:", id);
        const { actions } = this.props;
        const row = getObjectByKey(this.state.data, 'id', id);
        const obj = row ? row.toJS() : {};
        // const data = {
        //     playerId: obj.id,
        //     playerName: obj.name,
        //     width: obj.width,
        //     height: obj.height
        // }
        //
        // actions.overlayerShow(<PlayerListPopup title="添加方案" data={data}
        //                                        onCancel={()=>{actions.overlayerHide()}} onConfirm={()=>{
        //      actions.overlayerHide();
        this.props.router.push({
            pathname: "/mediaPublish/playerArea",
            state: { item: obj }
        });
        // }}/>)
    }

    removeHandler(id) {
        const { actions } = this.props;
        actions.overlayerShow(<ConfirmPopup iconClass="icon_popup_delete" tips={this.props.intl.formatMessage({id:'mediaPublish.isDeleteList'})}

            cancel={() => { actions.overlayerHide() }} confirm={() => {
            }} />);
    }

    render() {
        const { type, search, page, data } = this.state;
        return <Content className="player-list">
            <div className="heading">
                <Select className="type" data={type}
                    onChange={(selectIndex) => this.typeChange(selectIndex)} />
                <SearchText placeholder={search.get('placeholder')} value={search.get('value')}
                    onChange={this.searchChange} submit={this.searchSubmit} />
                <button className="btn btn-primary add-playerList" onClick={() => this.addHandler()}><FormattedMessage id='button.add' /></button>
            </div>
            <div className="playerList-container">
                <Table columns={this.columns} data={data} isEdit={true} rowEdit={this.editHandler} rowDelete={this.removeHandler} />
                <Page className={"page " + (page.get('total') == 0 ? "hidden" : "")} showSizeChanger pageSize={page.get('pageSize')}
                    current={page.get('current')} total={page.get('total')} onChange={this.pageChange} />
            </div>

        </Content>
    }
}

const mapStateToProps = state => {
    return {
        sidebarNode: state.mediaPublish.get('sidebarNode')
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators({
            overlayerShow: overlayerShow,
            overlayerHide: overlayerHide,
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(injectIntl(PlayerList))
