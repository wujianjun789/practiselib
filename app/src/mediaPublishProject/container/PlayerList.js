/**
 * Created by a on 2017/10/17.
 */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Content from '../../components/Content';
import Select from '../../components/Select';
import SearchText from '../../components/SearchText';
import Table from '../../components/Table';
import Page from '../../components/Page';

import PlayerListPopup from '../component/PlayerListPopup';
import ConfirmPopup from '../../components/ConfirmPopup';
import NotifyPopup from '../../common/containers/NotifyPopup';

import {searchProjectList, getProjectByName, addProject, updateProjectById, removeProjectById, projectPublish} from '../../api/mediaPublish';
import {applyClick} from '../action/index';

import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';
import {addNotify, removeAllNotify} from '../../common/actions/notifyPopup';
import Immutable from 'immutable';
import { getObjectByKey } from '../../util/algorithm';

import { FormattedMessage, injectIntl } from 'react-intl';

export class PlayerList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      type: Immutable.fromJS({
        list: [{ id: 1, value: `${this.formatIntl('mediaPublish.playType')} 1` },
          { id: 2, value: `${this.formatIntl('mediaPublish.playType' )} 2` }],
        index: 0, value: `${this.formatIntl('mediaPublish.playType')} 1`,
      }),
      search: Immutable.fromJS({ placeholder: this.formatIntl('mediaPublish.inputSchemeName'), value: '' }),
      page: Immutable.fromJS({
        pageSize: 10,
        current: 1,
        total: 0,
      }),
      data: Immutable.fromJS([
        /* { id: 1, icon: "", name: `${this.formatIntl('mediaPublish.playPlan')} 1`, resolution: "800X600", width: 800, height: 600 },
                { id: 2, icon: "", name: `${this.formatIntl('mediaPublish.playPlan')} 2`, resolution: "800X600", width: 800, height: 600 }*/
      ]),
    };

    this.columns = [{ id: 1, field: 'name', title:this.formatIntl('mediaPublish.schemeName')},
      { id: 2, field: 'resolution', title: this.formatIntl('mediaPublish.resolution')}];
    this.publishResponse = false;

    this.formatIntl = this.formatIntl.bind(this);

    this.typeChange = this.typeChange.bind(this);
    this.searchChange = this.searchChange.bind(this);
    this.searchSubmit = this.searchSubmit.bind(this);
    this.pageChange = this.pageChange.bind(this);
    this.updatePage = this.updatePage.bind(this);

    this.addHandler = this.addHandler.bind(this);
    this.funHandler = this.funHandler.bind(this);
    this.renameHandler = this.renameHandler.bind(this);
    this.publishHandler = this.publishHandler.bind(this);
    this.editHandler = this.editHandler.bind(this);
    this.removeHandler = this.removeHandler.bind(this);

    this.requestSearch = this.requestSearch.bind(this);
    this.updateSearch = this.updateSearch.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.requestSearch();
  }

  componentWillUnmount() {
    this.mounted = false;
    this.props.actions.removeAllNotify();
  }

  formatIntl(formatId) {
    const {intl} = this.props;
    return intl ? intl.formatMessage({id:formatId}) : null;
    // return formatId;
  }

  requestSearch() {
    const {page, search} = this.state;
    const limit = page.get('pageSize');
    const offset = (page.get('current') - 1) * limit;
    const searchName = search.get('value');

    getProjectByName(0, searchName, data => {this.mounted && this.updatePageTotal(data);});
    searchProjectList(0, searchName, offset, limit, data => {this.mounted && this.updateSearch(data);});
  }

  updatePageTotal = (data) => {
    this.setState({page:this.state.page.update('total', v => data.length)});
  }

  updateSearch(data) {
    data = data.map(project => {
      return Object.assign({}, project, {'resolution':project.width + 'X' + project.height});
    });

    this.setState({data:Immutable.fromJS(data)});
  }

  typeChange(selectIndex) {
    this.state.type = this.state.type.update('index', v => selectIndex);
    this.setState({ type: this.state.type.update('value', v => this.state.type.getIn(['list', selectIndex, 'value'])) }, () => {
      this.requestSearch();
    });
  }

  searchChange(value) {
    this.setState({ search: this.state.search.update('value', v => value) });
  }

  searchSubmit() {
    this.updatePage(1);
  }

  pageChange(current, pageSize) {
    this.updatePage(current);
  }

  updatePage(current) {
    this.setState({ page: this.state.page.update('current', v => current) }, () => {
      this.requestSearch();
    });
  }

  addHandler() {
    const { actions } = this.props;
    const data = {
      id: '',
      name: '',
      width: 800,
      height: 600,
    };

    actions.overlayerShow(<PlayerListPopup intl={this.props.intl} title={this.formatIntl('mediaPublish.addPlan')} data={data}

      onCancel={() => { actions.overlayerHide(); }} onConfirm={(state) => {

        let data = {
          name: state.name,
          width: state.width,
          height: state.height,
        };

        addProject(data, (response) => {
          actions.overlayerHide();
          this.props.router.push({
            pathname: '/mediaPublish/playProject/' + response.id
          });
        });
      }} />);

  }

  funHandler(id) {

  }

  renameHandler(id){
    const { actions } = this.props;
    const row = getObjectByKey(this.state.data, 'id', id);
    const obj = row ? row.toJS() : {};
    const data = {
      id: obj.id,
      name: obj.name,
      width: obj.width,
      height: obj.height,
    };

    actions.overlayerShow(<PlayerListPopup IsModify={true} intl={this.props.intl} title={this.formatIntl('mediaPublish.modify')} data={data}
                                           onCancel={() => { actions.overlayerHide(); }} onConfirm={(state) => {
                                              // actions.applyClick("playerProject");
                                              const newData = {
                                                id: state.id,
                                                name: state.name,
                                                width: state.width,
                                                height: state.height,
                                              };
                                              updateProjectById(newData, ()=>{
                                                this.requestSearch();
                                              })
                                           }}/>)
  }

  publishHandler(id){
    const {actions} =this.props;
    if(this.publishResponse){
      return actions.addNotify(0, "请等待发布完成。");
    }
    this.publishResponse = true;
    projectPublish(id, ()=>{
      this.publishResponse = false;
      actions.addNotify(1, "发布成功。");
    });
  }

  editHandler(id) {
    const { actions } = this.props;
    const row = getObjectByKey(this.state.data, 'id', id);
    const obj = row ? row.toJS() : {};

    this.props.router.push({
      pathname: '/mediaPublish/playProject/' + obj.id
    });
  }

  removeHandler(id) {
    const { actions } = this.props;
    actions.overlayerShow(<ConfirmPopup iconClass="icon_popup_delete" tips={this.formatIntl('mediaPublish.isDeleteList')}
      cancel={() => { actions.overlayerHide(); }} confirm={() => {
        removeProjectById(id, () => {
          actions.overlayerHide();
          this.requestSearch();
        });
      }} />);
  }

  render() {
    const { type, search, page, data } = this.state;
    return <Content className="player-list">
      <div className="heading">
        <Select className="type" data={type}
          onChange={(selectIndex) => this.typeChange(selectIndex)} />
        <SearchText placeholder={search.get('placeholder')} value={search.get('value')}
          onChange={this.searchChange} submit={this.searchSubmit} />
        <button className="btn btn-primary add-playerList" onClick={() => this.addHandler()}><FormattedMessage id="button.add" /></button>
      </div>
      <div className="playerList-container">
        <Table columns={this.columns} data={data} isProject={true} isEdit={true} rowRename={this.renameHandler} rowPublish={this.publishHandler} rowEdit={this.editHandler} rowDelete={this.removeHandler} />
        <Page className={'page ' + (page.get('total') == 0 ? 'hidden' : '')} showSizeChanger pageSize={page.get('pageSize')}
          current={page.get('current')} total={page.get('total')} onChange={this.pageChange} />
      </div>
      <NotifyPopup/>
    </Content>;
  }
}

const mapStateToProps = state => {
  return {
    sidebarNode: state.mediaPublishProject.sidebarNode
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators({
      overlayerShow: overlayerShow,
      overlayerHide: overlayerHide,
      applyClick: applyClick,
      addNotify: addNotify,
      removeAllNotify : removeAllNotify
    }, dispatch),
  };
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(PlayerList));
