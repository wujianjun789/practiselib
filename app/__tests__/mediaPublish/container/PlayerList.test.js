/**
 * Created by a on 2017/10/18.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import PlayerList from '../../../src/mediaPublish/container/PlayerList';


test('DomainEdit renders', ()=>{
    const component = renderer.create(
        <PlayerList />
    )

    let playerList = component.toJSON();
    expect(playerList).toMatchSnapshot();
})

test('DomainEdit div click', ()=>{
    const component = shallow(<PlayerList />)
    expect(component.find('.heading').length).toEqual(1);
    expect(component.find('.playerList-container').length).toEqual(1);
})