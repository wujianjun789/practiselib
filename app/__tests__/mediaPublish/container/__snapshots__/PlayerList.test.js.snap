exports[`test PlayerList renders 1`] = `
<div
  className="content player-list">
  <div
    className="heading">
    <select
      className="select type form-control"
      id=""
      onChange={[Function]}
      placeholder={undefined}
      value="播放类型1">
      <option
        value="播放类型1">
        播放类型1
      </option>
      <option
        value="播放类型2">
        播放类型2
      </option>
    </select>
    <div
      className="searchText "
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="输入方案名称"
        type="search"
        value="" />
      <ul
        className="" />
      <span
        className="glyphicon glyphicon-search"
        onClick={[Function]} />
    </div>
    <button
      className="btn btn-primary add-playerList"
      onClick={[Function]}>
      添加
    </button>
  </div>
  <div
    className="playerList-container">
    <div
      className="table-responsive ">
      <table
        className="table table-hover">
        <thead>
          <tr>
            <th
              className="hidden">
              <input
                checked={undefined}
                onChange={[Function]}
                type="checkbox" />
            </th>
            <th>
              方案名称
            </th>
            <th>
              分辨率
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
            onClick={[Function]}>
            <td
              className="hidden">
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox" />
            </td>
            <td>
              播放方案1
            </td>
            <td>
              1920X1080
            </td>
            <td
              className="edit">
              <a
                className="btn"
                onClick={[Function]}>
                <span
                  className="icon_edit" />
                <span
                  className="update">
                  修改
                </span>
              </a>
              <a
                className="btn"
                onClick={[Function]}>
                <span
                  className="icon_delete" />
                <span
                  className="del">
                  删除
                </span>
              </a>
            </td>
          </tr>
          <tr
            className=""
            onClick={[Function]}>
            <td
              className="hidden">
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox" />
            </td>
            <td>
              播放方案2
            </td>
            <td>
              1920X1080
            </td>
            <td
              className="edit">
              <a
                className="btn"
                onClick={[Function]}>
                <span
                  className="icon_edit" />
                <span
                  className="update">
                  修改
                </span>
              </a>
              <a
                className="btn"
                onClick={[Function]}>
                <span
                  className="icon_delete" />
                <span
                  className="del">
                  删除
                </span>
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <ul
      className="rc-pagination page "
      style={Object {}}
      unselectable="unselectable">
      <li
        aria-disabled={true}
        className="rc-pagination-disabled rc-pagination-prev"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title="上一页">
        <a>
          
        </a>
      </li>
      <li
        className="rc-pagination-item rc-pagination-item-1 rc-pagination-item-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}>
        <a>
          1
        </a>
      </li>
      <li
        aria-disabled={true}
        className="rc-pagination-disabled rc-pagination-next"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title="下一页">
        <a>
          
        </a>
      </li>
      <div
        className="rc-pagination-options" />
    </ul>
  </div>
</div>
`;
